src/test/pages/TeacherRequest.test.tsx

```diff
@@ -0,0 +1,204 @@
+import { TestUserRole, checkPageTitle, renderTestEnvironment } from "../utils";
+import { TeacherRequest } from "../../app/components/pages/TeacherRequest";
+import { RestHandler, rest } from "msw";
+import { API_PATH } from "../../app/services";
+import { screen } from "@testing-library/react";
+import { produce } from "immer";
+import { mockUser } from "../../mocks/data";
+import userEvent from "@testing-library/user-event";
+import * as actions from "../../app/state/actions";
+
+function checkDisabledAndCorrectValue(input: HTMLElement, expectedValue: string) {
+  expect(input).toHaveValue(expectedValue);
+  expect(input).toBeDisabled();
+}
+
+const upgradeAccountSpy = jest.spyOn(actions, "upgradeAccount");
+
+describe("TeacherRequest", () => {
+  const setupTest = ({
+    role,
+    extraEndpoints,
+    modifyUser,
+  }: {
+    role: TestUserRole;
```

Comment:
`UserRole`? Since this will always be a student or a teacher, right?

Reply:
In these particular tests yes, although arguably I should probably implement another test that checks what shows for other user roles? (it should be the same as what shows for teacher)

Reply:
I wouldn't add a test like that, because at the moment our business logic says that it's only possible to request an upgrade from a student. So you'd be adding a new test for functionality that we don't need, and it's not a valid use case (at least at the moment).